### DISCUSSION TITLE
Can't change `desc` of default LSP keymap to group name

### DISCUSSION THREAD

**Discussion by coder9194** [LazyVim/LazyVim#3754] [Q&A] [answered]
I want to overwrite some LSP keymaps to support keymaps in groups, e.g. "Ke" for showing error diagnostic details in popup under current hover, "gdh" for goto definition on left window.
So I follow the guide here https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps and set things like
```lua
-- nvim-lspconfig.lua
{
  "neovim/nvim-lspconfig",
  opts = function()
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    -- change a keymap
    keys[#keys + 1] = { "K", false }
    keys[#keys + 1] = { "gd", false }
  end,
}

-- keymaps.lua
require("which-key").register({
  gd = { name = "+goto definition" },
  K = { name = "+hover" },
})
```

But I can't find a way to change the `desc` shown in `which-key.nvim`. Is there anything I am missing in the doc or in old disscussion threads?

**Comment by iamxiaojianzheng** [comment]
It works for me.

![image](https://github.com/LazyVim/LazyVim/assets/29248764/5e91c7a3-1733-4c24-b6f3-14c3fc763aad)

**Reply by coder9194** [reply]
Sorry for my unclear question, I updated OP.

**Comment by dpetka2001** [comment]
Just add a `desc = "My custom K"` in the mapping as well?

**Reply by coder9194** [reply]
Sorry for my unclear question, I updated OP.

**Reply by dpetka2001** [reply]
Just do
```lua
{
  "neovim/nvim-lspconfig",
  opts = function()
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    -- change a keymap
    keys[#keys + 1] = { "K", "", desc = "+hover" }
    keys[#keys + 1] = { "gd", "", desc = "+goto definition" }
  end,
}
```
Notice that the `rhs` is just an empty string in both cases `""`.

**Reply by coder9194** [reply]
Never thought I can do the key overwriting in opts function lol.
Now I can change the `desc` to become group, but still can't change the text of `desc` shown in `which-key.nvim`.
![image](https://github.com/LazyVim/LazyVim/assets/46557970/aba0f307-55ab-4261-a42b-3ba5b9c9228d)

Anyway I just want to change keymap to be a group. So problem solved & thank you!

**Reply by dpetka2001** [reply, marked as answer]
I did the following
```lua
      local Keys = require("lazyvim.plugins.lsp.keymaps").get()
      vim.list_extend(Keys, {
        { "gd", "", desc = "+my custom gd" },
        { "gdl", "<cmd>FzfLua lsp_definitions<cr>", desc = "Go to definition" },
      })
```
and the description of `gd` changed successfully.

**Reply by coder9194** [reply]
Wow, works like a charm now! Thank you so much!
